test/parallel/test-module-hooks-import-wasm.mjs

```diff
@@ -0,0 +1,35 @@
+// Flags: --no-experimental-wasm-modules
```

Comment:
Please put all the new tests under `test/es-module/`. It’s been our practice for years to run the modules tests as a subset via `tools/test.py --mode=release test/es-module`. At this point perhaps `es-module` should just be named `modules`, but it’s very useful to be able to run all the modules-related tests in a few seconds without needing to run the full `test/parallel` set of tests.

Reply:
I don't think they should be put under `test/es-module`, because they are not ESM specific. Though I could add them to `test/modules` (not what I would recommend, because then you would start moving `test/es-modules` to `test/modules` and make backporting harder), or `test/module-hooks` (less impactful so would be okay-ish for backporting).

> It’s been our practice for years to run the modules tests as a subset via tools/test.py --mode=release test/es-module

I don't think that's the right way to "run all tests related to modules" in the first place, because you would be missing a lot of CJS tests with this. Perhaps you meant it's a practice to run "tests that are only related to ESM"? Then that doesn't apply to the tests added here, because they are not ESM-only.

> but it’s very useful to be able to run all the modules-related tests in a few seconds without needing to run the full test/parallel set of tests.

In case you didn't know, you can run `tools/test.py "test/parallel/test-module-hook*"`. If you want a shortcut to run all module-related tests, I would suggest to add a Makefile target that runs `tools/test.py es-module`, `tools/test.py test/parallel/*module*` which would provide proper coverage, instead of counting on `tools/test.py es-module` to cover all the module tests, since it already doesn't do that.

Reply:
I think in general we shouldn’t be dumping more and more tests into `test/parallel`. It’s already unusable in a file explorer, having thousands of files within it. If you don’t want to put them in `test/es-module` then perhaps this can be the PR where we create `test/module`, and a follow-up can migrate `test/es-module` into it?

Reply:
> a follow-up can migrate test/es-module into it?

I feel that's going to create a lot of headaches for backports (which I absolutely won't enjoy when I backport `require(esm)` to v20..). I would prefer `test/module-hooks` for a smaller scope.

Reply:
> I would prefer `test/module-hooks` for a smaller scope.

I was thinking maybe that `test/*` could mirror `lib/*` and `src/*`, where we have `lib/modules`, but I guess it can't hurt to be even more specific.

Reply:
I feel that there are many mistakes made in the `lib/something` and `src/something` naming choices (some by myself which I know regret about as they just make the file harder to find for me, as the category may not really make too much sense)